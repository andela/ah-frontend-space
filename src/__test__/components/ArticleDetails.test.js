import React from 'react';
import { shallow } from 'enzyme';
import HomeDetails, { Interest } from '../../components/HomePage/articleDetails';

const props2 = {
  article: {
    author: {
      username: 'james',
    },
    created_at: '',
    body: '',
    read_time: '',
    likes_count: '',
    dislikes_count: '',
    title: '',
    image: 'http://myimage.jpg',
  },
};

describe('App', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<HomeDetails {...props2} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render interest', () => {
    const props = {
      klass: 'likes',
      alternative: 'likes',
      src: 'http://myimage.jpg',
      value: '5',
    };
    const wrapper = shallow(<Interest {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render interest', () => {
    const props = {
      klass: 'dislikes',
      alternative: 'likes',
      src: 'http://myimage.jpg',
      value: '5',
    };
    const wrapper = shallow(<Interest {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    const props3 = {
      article: {
        author: {
          username: 'james',
        },
        created_at: '',
        body: '',
        read_time: '',
        likes_count: '',
        dislikes_count: '',
        title: '',
        image: null,
      },
    };
    const wrapper = shallow(<HomeDetails {...props3} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    sessionStorage.setItem('username', 'james');
    const wrapper = shallow(<HomeDetails {...props2} />);
    expect(wrapper).toMatchSnapshot();
  });
});
