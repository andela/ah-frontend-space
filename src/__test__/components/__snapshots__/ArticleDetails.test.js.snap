// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App it should render interest 1`] = `ShallowWrapper {}`;

exports[`App it should render interest 2`] = `ShallowWrapper {}`;

exports[`App should match snapshot 1`] = `ShallowWrapper {}`;

exports[`App should match snapshot 2`] = `ShallowWrapper {}`;

exports[`App should match snapshot 3`] = `ShallowWrapper {}`;
